### PR TITLE
skip symlink

### PR DESCRIPTION
**Pull request by gaodeng**

### CODE REVIEW

**Comment by mikehardy** [comment]
@gaodeng this looks fantastic, thank you! I have one suggestion based on the CI results - will make it in separate comment in a second

File: src/utils.js [hunk 1]
```diff
@@ -11,10 +11,13 @@ const chunk = (array, chunkSize) => {
 };
```
Comment (mikehardy) [inline src/utils.js]:
```suggestion
const nodeVerMajor = process.version.split('.')[0];
const nodeVerMinor = process.version.split('.')[1];
const symLinkSupported = (nodeVerMajor > 10) || (nodeVerMajor == 10 && nodeVerMinor >= 10);
```

File: src/utils.js [hunk 2]
```diff
@@ -11,10 +11,13 @@ const chunk = (array, chunkSize) => {
...
 
 const readDir = (dir, filesList = []) => {
-  const files = readdirSync(dir);
-  for (let file of files) {
+  const direntList = readdirSync(dir, { withFileTypes: true });
+  for (let dirent of direntList) {
+    if (dirent.isSymbolicLink()) {
```
Comment (mikehardy) [inline src/utils.js]:
```suggestion
    if (symLinkSupported && dirent.isSymbolicLink()) {
```

**Comment by mikehardy** [comment]
That looks cleaner :-), thank you again - just waiting on CI to chew through it

**Comment by gaodeng** [comment]
> It looks like the method I requested you use is only available on node 10.10 and up, I made an untested suggestion that should guard against that, what do you think?

thanks for your hints. I switched to using lstatSync and hope it will pass the test now

**Comment by mikehardy** [comment]
released :rocket:  https://github.com/mikehardy/jetifier/releases/tag/v1.6.5

**Pull request merged**